### PR TITLE
Update project format and packaging

### PR DESCRIPTION
**Pull request by holytshirt**
Hi there!

Thank you so much for contributing this pull request. We love when the community
steps up and share, and we're enormously thankful.

You don't have to do all of those following things, they are what we try and get 
done for code coming in:

 - [x] the code
 - [x] If it's a non-trivial piece of code, signing a CLA
 - [ ] If it breaks, change, fix or add to existing behaviour, updating
       CHANGELOG.md
 - [ ] If it's a bug fix, tests in the Tests `project`, or scenarios in the `TestRig`.
 - [x] On top of HEAD, unless it's a bugfix on a previous version
 - [ ] VERSION file updated if not creating a pull-request on top of `master`

If those are needed, and you haven't and can't, we thank you enormously for your
contribution, and we'll add those before merging the pull request.

Thanks!

The OpenRasta community

### CODE REVIEW

File: .github/workflows/dotnet.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      with:
+        dotnet-version: 5.0.x
+    - name: Restore dependencies
+      run: dotnet restore
+    - name: Build
+      run: dotnet build --no-restore
+    - name: Test
+      run: dotnet test --no-build --verbosity normal
+    - name: Upload test results
+      if: success() || failure()
```
Comment (serialseb) [inline .github/workflows/dotnet.yml:30]:
```suggestion
      if: always()
```

**Comment by serialseb** [comment]
While we're doing a bit of upkeep and maintenance, would it be worth getting rid of unsupported net46 builds if they're EOL?

**Comment by holytshirt** [comment]
> While we're doing a bit of upkeep and maintenance, would it be worth getting rid of unsupported net46 builds if they're EOL?

@serialseb okay I'll push to net472, better compatibility with .netstandard2.0

**Comment by holytshirt** [comment]
@serialseb This is ready now, includes updating to net6, package updates, build done by github actions.
Packages are pushed Appveyor too

**Pull request merged**